rotkehlchen/tests/unit/test_globaldb_asset_updates.py

```diff
@@ -295,3 +296,45 @@ def test_parse_full_insert(assets_updater, text, expected_data, error_msg):
+    """
+    Checks the folowing cases:
+    1. If some of the updates are broken, others are not affected.
+    2. If an update is broken, info about asset stays the same in all tables as before the update.
+
+    Checks both insertion of a new asset and an update of an existing one.
+    """
+    # In "ETH" update swapped_for is incorrect
+    # In "BTC" update everything is good
+    # In "NEW-ASSET-1" insertion swapped_for is incorrect
+    # In "NEW-ASSET-2" insertion everything is good
+    update_text = """INSERT INTO assets(identifier, name, type) VALUES("ETH", "name1", "B"); INSERT INTO common_asset_details(identifier, symbol, coingecko, cryptocompare, forked, started, swapped_for) VALUES("ETH", "symbol1", "", "", NULL, NULL, "NONEXISTENT");
+*
+INSERT INTO assets(identifier, name, type) VALUES("BTC", "name2", "B"); INSERT INTO common_asset_details(identifier, symbol, coingecko, cryptocompare, forked, started, swapped_for) VALUES("BTC", "symbol2", "", "", NULL, NULL, NULL);
+*
+INSERT INTO assets(identifier, name, type) VALUES("NEW-ASSET-1", "name3", "B"); INSERT INTO common_asset_details(identifier, symbol, coingecko, cryptocompare, forked, started, swapped_for) VALUES("NEW-ASSET-1", "symbol3", "", "", NULL, NULL, "LOLKEK");
+*
+INSERT INTO assets(identifier, name, type) VALUES("NEW-ASSET-2", "name4", "B"); INSERT INTO common_asset_details(identifier, symbol, coingecko, cryptocompare, forked, started, swapped_for) VALUES("NEW-ASSET-2", "symbol4", "", "", NULL, NULL, NULL);
+*
+"""  # noqa: E501
+    connection = GlobalDBHandler().conn
+    assets_updater._apply_single_version_update(
+        connection=connection,
+        version=999,  # doesn't matter
+        text=update_text,
+        conflicts={'ETH': 'remote', 'BTC': 'remote'},
+    )
+
+    # Should have stayed unchanged since one of the insertions was incorrect
+    assert connection.execute('SELECT name FROM assets WHERE identifier="ETH"').fetchone()[0] == 'Ethereum'  # noqa: E501
+    assert connection.execute('SELECT symbol FROM common_asset_details WHERE identifier="ETH"').fetchone()[0] == 'ETH'  # noqa: E501
+    # BTC should have been updated since the insertion were correct
+    assert connection.execute('SELECT name FROM assets WHERE identifier="BTC"').fetchone()[0] == 'name2'  # noqa: E501
+    assert connection.execute('SELECT symbol FROM common_asset_details WHERE identifier="BTC"').fetchone()[0] == 'symbol2'  # noqa: E501
+    # NEW-ASSET-1 should have not been added since the insertions were incorrect
+    assert connection.execute('SELECT * FROM assets WHERE identifier="NEW-ASSET-1"').fetchone() is None  # noqa: E501
+    # NEW-ASSET-2 should have been added since the insertions were correct
+    assert connection.execute('SELECT name FROM assets WHERE identifier="NEW-ASSET-2"').fetchone()[0] == 'name4'  # noqa: E501
+    assert connection.execute('SELECT symbol FROM common_asset_details WHERE identifier="NEW-ASSET-2"').fetchone()[0] == 'symbol4'  # noqa: E501
```

Comment:
nice but as I mentioned in the example you need specific unit tests of the save functionality you made. You need to test that `with db.savepoint_ctx()` works properly for multiple scenarios, and also offer non context manager functions to roll back to savepoints.

Generally a proper driver/api needs to exist for this in `DBConnection` and be thoroughly unit tested (in the proper test file)